pkg/api/handlers/compat/images.go

```diff
@@ -381,19 +381,22 @@ func GetImage(w http.ResponseWriter, r *http.Request) {
 	// 404 no such
 	// 500 internal
 	name := utils.GetName(r)
-	possiblyNormalizedName, err := utils.NormalizeToDockerHub(r, name)
+	newImage, err := utils.GetImage(r, name)
```

Comment:
We must normalize _all_ input to the Docker endpoints. A short-name would otherwise not be normalized to docker.io.

Reply:
The issue I am seeing is it is getting screwed up with it looks for sha256:UUID.
It is getting normalized to:

docker.io/library/sha256:c059bfaa849c4d8e4aecaeb3a10c2d9b3d85f5165c66ad3a4d937758128c4d18

Reply:
We need to update `NormalizeToDockerHub` to handle that but no caller to `NormalizeToDockerHub` should be changed.

Reply:
I _think_ that if it starts with `sha256:`, the name should be returned as is.

Reply:
SGTM
Could you fix this, since I am supposed to be on PTO.

Reply:
Sure.

Reply:
https://github.com/containers/podman/pull/12663